Assets/Scripts/Combat/PlayerCombat.cs

```diff
@@ -0,0 +1,52 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PlayerCombat : Entity {
+
+    [SerializeField, Tooltip("The range of the player attack as a float")]
+    float _attackRange;
+    [SerializeField, Tooltip("How often the player can attack")]
+    float _attackRate;
+
+    float _nextAttackTime = 0f;
+
+    public Animator animator;
+    public Transform attackPoint;
+    public LayerMask enemyLayers;
+
+    // Update is called once per frame
+    void Update()
+    {
+        if (Time.time >= _nextAttackTime) {
+            if (Input.GetKeyDown(KeyCode.X)) {
```

Comment:
It's a good idea to use Input.GetButtonDown here instead and setup a new Button in the InputManager (Edit -> Project Settings -> Input Manager), because it allows the player to change the keybinding if they want to.